programs-remote-listings/rs-connect.php

```diff
@@ -63,6 +65,21 @@ public function remove_rel_canonical()
         }
     }
 
+    public function verify_listing_exists() {
+        $program_id = (int) get_query_var('rs_program');
+        $teacher_id = (int) get_query_var('rs_teacher');
+        $program = $program_id ? RS_Connect_Api::get_program($program_id) : false;
+        $teacher = $teacher_id ? RS_Connect_Api::get_teacher($teacher_id) : false;
```

Comment:
given this plugin might be installed on client websites with ancient versions of php running, should I use `if (empty())` rather than a ternary `?:`

Reply:
eg

```$program_id = (int) get_query_var('rs_program');
        $teacher_id = (int) get_query_var('rs_teacher');
        $program = null;
        $teacher = null;
        
        if ($program_id) {
            $program = RS_Connect_Api::get_program($program_id);

            if (! $program) {
                self::force_404();
            }
        }
        
        if ($teacher_id) {
            $teacher = RS_Connect_Api::get_teacher($teacher_id);

            if (! $teacher) {
                self::force_404();
            }
        }```

Reply:
get rid of the $teacher and $program vars and then this is actually cleaner code. ternaries might result in short code, but not always better code.